9.2/root/usr/share/container-scripts/postgresql/common.sh

```diff
@@ -1,6 +1,22 @@
 # Configuration settings.
 export POSTGRESQL_MAX_CONNECTIONS=${POSTGRESQL_MAX_CONNECTIONS:-100}
-export POSTGRESQL_SHARED_BUFFERS=${POSTGRESQL_SHARED_BUFFERS:-32MB}
+
+# Perform auto-tuning based on the container cgroups limits (only when the
+# limits are set).
+# Users can still override this by setting the POSTGRESQL_SHARED_BUFFERS
+# and POSTGRESQL_EFFECTIVE_CACHE_SIZE variables.
+if [[ "${NO_MEMORY_LIMIT}" == "true" ]]; then
+    export POSTGRESQL_SHARED_BUFFERS=${POSTGRESQL_SHARED_BUFFERS:-32MB}
+    export POSTGRESQL_EFFECTIVE_CACHE_SIZE=${POSTGRESQL_EFFECTIVE_CACHE_SIZE:-128MB}
+else
+    # Use 1/4 of given memory for shared buffers
+    shared_buffers_computed="$(python -c "print(int((${MEMORY_LIMIT_IN_BYTES}/(1024*1024))*0.25))")M"
```

Comment:
After I accidentally commited `cgroup-limits` file as non-executable I had an error
```
/usr/bin/run-mysqld: line 3: /usr/bin/cgroup-limits: Permission denied
  File "<string>", line 1
    print(int((/(1024*1024))*0.1))
               ^
SyntaxError: invalid syntax
```
So I also added additional check for emptiness of `MEMORY_LIMIT_IN_BYTES`. Maybe you also would like to add it here. Just to be sure.